components/engine_rocks/src/config.rs

```diff
@@ -263,51 +262,76 @@ impl From<BlobRunMode> for DBTitanDBBlobRunMode {
     }
 }
 
-macro_rules! numeric_enum_mod {
+macro_rules! impl_enum_with_mirror_type_mod {
     ($name:ident $enum:ident { $($variant:ident = $value:expr, )* }) => {
         pub mod $name {
             use std::fmt;
 
             use serde::{Serializer, Deserializer};
             use serde::de::{self, Unexpected, Visitor};
-            use rocksdb::$enum;
+            use rocksdb::$enum as T;
+            use case_macros::*;
 
-            pub fn serialize<S>(mode: &$enum, serializer: S) -> Result<S::Ok, S::Error>
+            enum $enum {
+                $($variant = $value, )*
+            }
+
+            impl From<$enum> for T {
+                fn from(m: $enum) -> Self {
+                    match m {
+                        $( $enum::$variant => T::$variant, )*
+                    }
+                }
+            }
```

Comment:
Oh, seems I was wrong. We don't actually need this anymore. Because we don't actually use the mirror type in other modules.

Reply:
Nice, i will reduce the code again